controlplane/pkg/prefix_pool/prefixpool.go

```diff
@@ -221,24 +250,29 @@ func ExtractPrefix(prefixes []string, prefixLen uint32) (string, []string, error
 			// We found required one.
 			resultPrefix := prefixes[idx]
 			resultPrefixes := append(prefixes[:idx], prefixes[idx+1:]...)
+			/* Verify the allocated addresses does not intersect with one from the excludedPrefix list */
+			err := verifyExcluding(resultPrefix, excludedPrefixes)
+			if err != nil {
+				return "", prefixes, err
+			}
 			// Lets remove from list and return
 			return resultPrefix, resultPrefixes, nil
 		} else {
 			// Update minimal root for split
-			if uint32(parentLen) < prefixLen && (parentLen > max_prefix || max_prefix == 0) {
-				max_prefix = parentLen
-				max_prefix_idx = idx
+			if uint32(parentLen) < prefixLen && (parentLen > maxPrefix || maxPrefix == 0) {
+				maxPrefix = parentLen
+				maxPrefixIndex = idx
 			}
 		}
 	}
 	// Not found, lets split minimal found prefix
-	if max_prefix_idx == -1 {
+	if maxPrefixIndex == -1 {
 		// There is no room to split
 		return "", prefixes, fmt.Errorf("Failed to find room to have prefix len %d at %v", prefixLen, prefixes)
 	}
 
-	resultPrefixRoot := prefixes[max_prefix_idx]
-	right_parts := []string{}
+	resultPrefixRoot := prefixes[maxPrefixIndex]
+	rightParts := []string{}
 
 	_, rootCIDRNet, _ := net.ParseCIDR(resultPrefixRoot)
 	for {
```

Comment:
Here we need to check and choose a direction we need to split based on 
verifyExcluding(resultPrefix, excludedPrefixes) to select left or right subtree, so we will not split just to left one.